src/main/resources/public/js/components/rightPane.js

```diff
@@ -0,0 +1,258 @@
+            const otherRec = JSON.parse(JSON.stringify(this.$store.state.currentRec));
+            otherRec.name = this.rename;
+            $.post({
+                url: 'recipes',
+                data: JSON.stringify(otherRec),
+                contentType: 'application/json',
+                dataType: 'json',
+            }).done(
+                (data) => {
+                    this.$store.commit('updateCurRec', data);
+                    this.$store.commit('sortRecList');
+                },
+            ).fail(
+                (jqXHR) => {
+                    this.$store.state.showRecTitle = true;
+                    this.showRename = false;
+                    this.fail(jqXHR);
+                },
+            );
+            this.showRename = false;
+            this.showRecTitle = true;
+        },
+        reset () {
+            this.showContainer = false;
+            this.$store.commit('updateCurRec', {});
+            // On page load, gets the recipe list, and appends as buttons to a ul in left pane.
+            $.get('/recipes', 'json')
+                .done(
+                    (recipeList) => {
+                        this.$store.commit('updateRecList', recipeList);
+                        this.$store.commit('sortRecList');
+                    },
+                ).fail(this.fail);
+
+            this.$emit('update-err', '');
+        },
+        showIng (idx) {
+            if (idx === this.editIngIdx) {
+                return false;
+            }
+            return true;
+        },
+        success (recipe) {
+            console.log('success!');
+            this.$store.commit('setCurRec', recipe);
+        },
+        delIng (idx) {
+            this.$store.state.currentRec.ingredients.splice(idx, 1)
+            $.ajax({
+                type: 'PUT',
+                url: `/recipes/${this.$store.state.currentRec.id}`,
+                data: JSON.stringify(this.$store.state.currentRec),
+                contentType: 'application/json',
+                dataType: 'json',
+            }).fail(this.fail)
+                .done(this.success);
+        }
+    },
+    /* html */
+    template: `
+      <div>
+          <div>
+              <strong id="rec-title" v-show="showRecTitle">{{this.$store.state.currentRec.name}}</strong>&nbsp;&nbsp;&nbsp;
+              <div id="renamed-recipe-name" v-show="showRename">
+                  <input id="rename-rec-input" placeholder="New Name..." type="text" v-model="rename" v-on:keyup.enter="applyRecRename">
+                  <button @click="applyRecRename" class="btn" id="apply-rec-rename"><img src="images/apply.png"></button>
+                  <button @click="cancelRecRename" class="btn" id="cancel-rec-rename"><img src="images/cancel.jpg"></button>
+              </div>
+              <button @click="editRecName" class="btn disabled" id="edit-rec-name" v-show="!showRename"><img
+                      src="images/edit.jpg"></button>
```

Comment:
Pull the `<img` down. Also, close the button on the next line rather than this one.

Reply:
Done.